example/resource.go

```diff
@@ -106,14 +109,11 @@ func metaProtocolRoute2HttpRoute(metaRoute metaroute.RouteConfiguration) *httpro
 	return httpRoute
 }
 
-func GenerateSnapshot() cache.Snapshot {
-	return cache.NewSnapshot(
-		"1",
-		[]types.Resource{}, // endpoints
-		[]types.Resource{}, // clusters
-		[]types.Resource{makeRoute()},
-		[]types.Resource{}, //listeners
-		[]types.Resource{}, // runtimes
-		[]types.Resource{}, // secrets
+func GenerateSnapshot() *cache.Snapshot {
+	snap, _ := cache.NewSnapshot("1",
+		map[resource.Type][]types.Resource{
+			resource.RouteType: {makeRoute()},
```

Comment:
I will write a complete example to demonstrate the meta_protocol_proxy of supporting the dubbo protocol in the future.